### PR TITLE
Fix a few narrowing conversions

### PR DESCRIPTION
**Pull request by EmmanuelMess**
Fix a few narrowing conversions

### CODE REVIEW

File: src/moodbar/moodbarbuilder.cpp [hunk 1]
```diff
@@ -187,9 +187,9 @@ QByteArray MoodbarBuilder::Finish(int width) {
 
     const int n = end - start;
 
-    *(data++) = rgb.r / n;
-    *(data++) = rgb.g / n;
-    *(data++) = rgb.b / n;
+    *(data++) = char(rgb.r / n);
```
Comment (jonaski) [inline src/moodbar/moodbarbuilder.cpp:190]: Please test this.
I did this previously, and the moodbar didn't work afterwards.
I reverted the change as part of https://github.com/strawberrymusicplayer/strawberry/commit/9cde537066c6332495e94c6bd2b4b7bc48ab32b8#diff-a20292def67aa3d74f22d1010c269c5420c5e0f93b758dea630c8b55eb4a2fff

Comment (jonaski) [reply src/moodbar/moodbarbuilder.cpp:190]: As far as I can see the change works here. So it is probably good.

**Comment by jonaski** [comment]
A couple of compile warnings when compiling with Clang, see the macOS builders:
```
In file included from /Users/runner/work/strawberry/strawberry/src/playlist/playlist.h:47:
/Users/runner/work/strawberry/strawberry/src/playlist/playlistitem.h:90:145: error: comparison of integers of different signs: 'quint64' (aka 'unsigned long long') and 'int' [-Werror,-Wsign-compare]
  qint64 effective_beginning_nanosec() const { return HasTemporaryMetadata() && temp_metadata_.is_valid() && temp_metadata_.beginning_nanosec() != -1 ? temp_metadata_.beginning_nanosec() : Metadata().beginning_nanosec(); }
                                                                                                             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ ^  ~~
/Users/runner/work/strawberry/strawberry/src/playlist/playlistitem.h:91:133: error: comparison of integers of different signs: 'quint64' (aka 'unsigned long long') and 'int' [-Werror,-Wsign-compare]
  qint64 effective_end_nanosec() const { return HasTemporaryMetadata() && temp_metadata_.is_valid() && temp_metadata_.end_nanosec() != -1 ? temp_metadata_.end_nanosec() : Metadata().end_nanosec(); }
                                                                                                       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~ ^  ~~
```

**Comment by EmmanuelMess** [comment]
> A couple of compile warnings when compiling with Clang, see the macOS builders:
> 
> ```
> In file included from /Users/runner/work/strawberry/strawberry/src/playlist/playlist.h:47:
> /Users/runner/work/strawberry/strawberry/src/playlist/playlistitem.h:90:145: error: comparison of integers of different signs: 'quint64' (aka 'unsigned long long') and 'int' [-Werror,-Wsign-compare]
>   qint64 effective_beginning_nanosec() const { return HasTemporaryMetadata() && temp_metadata_.is_valid() && temp_metadata_.beginning_nanosec() != -1 ? temp_metadata_.beginning_nanosec() : Metadata().beginning_nanosec(); }
>                                                                                                              ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ ^  ~~
> /Users/runner/work/strawberry/strawberry/src/playlist/playlistitem.h:91:133: error: comparison of integers of different signs: 'quint64' (aka 'unsigned long long') and 'int' [-Werror,-Wsign-compare]
>   qint64 effective_end_nanosec() const { return HasTemporaryMetadata() && temp_metadata_.is_valid() && temp_metadata_.end_nanosec() != -1 ? temp_metadata_.end_nanosec() : Metadata().end_nanosec(); }
>                                                                                                        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~ ^  ~~
> ```

Reverted offending commit, to fix this I would have to replace it with std::optional<quint64> instead, maybe another pr.

**Comment by jonaski** [comment]
Thanks!

**Comment by jonaski** [comment]
The change to `WordyTimeNanosec` caused a regression, the playlist showing incorrect length for songs without a length, like radio streams. But the bug was in `Playlist::GetTotalLength()` using quint64 instead of qint64..

**Comment by EmmanuelMess** [comment]
> The change to `WordyTimeNanosec` caused a regression, the playlist showing incorrect length for songs without a length, like radio streams. But the bug was in `Playlist::GetTotalLength()` using quint64 instead of qint64..

What should I do?

**Comment by jonaski** [comment]
Another regression related to the spotify cover provider refresh token:
`22:08:44.863 WARN  unknown                          QObject::startTimer: Timers cannot have negative intervals`

The following code where `quint64` was `qint64` before.

```
  if (!refresh_token_.isEmpty()) {
    quint64 time = expires_in_ - (QDateTime::currentDateTime().toSecsSinceEpoch() - login_time_);
    if (time < 1) time = 1;
    refresh_login_timer_.setInterval(static_cast<int>(time * kMsecPerSec));
    refresh_login_timer_.start();
  }
```

**Pull request merged**